meta-iotqa/lib/oeqa/runtime/connectivity/wifi/wifi.py

```diff
@@ -134,14 +134,23 @@ def check_internet_connection(self, url):
             time.sleep(3)
         self.target_collect_info("route")
         assert status == 0, "Error messages: %s" % self.log
-
     def execute_connection(self, ap_type, ssid, pwd):
         '''
-        Do a full round of wifi connection without disable
+        Do a full round of wifi connection without disable (try 3 times)
         '''
-        self.enable_wifi()
-        self.connect_wifi(ap_type, ssid, pwd)
-        self.wifi_ip_check()
+        for i in range(3):
+            try:
+                self.enable_wifi()
+                self.connect_wifi(ap_type, ssid, pwd)
+                self.wifi_ip_check()
+                break
+            except AssertionError:
+                if(i < 2):
```

Comment:
Even in C it would look weird.  `if` is not a function, but key word followed by an expression hence people put a whitespace between the key word and the rest of the statement to avoid confusions. 

Though I saw opposite coding styles - a whitespace between function name and brackets.